Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -1,50 +1,84 @@
-src\<Library Name>\pkg - Contains package projects for the library.
-src\<Library Name>\tests - Contains the test code for a library
-```
-In the src directory for a library there should be only **one** `.csproj` file that contains any information necessary to build the library in various configurations (see [Configurations](#project-configuration-conventions)). The src directory should also contain exactly **one** `.builds` file which contains all the valid configurations that a library should be built for (see [#.builds file](#project-builds-file)).
+#Build Project Guidelines
+In order to work in corefx repo you must first run build.cmd/sh from the root of the repo at least
+once before you can iterate and work on a given library project.
+
+##Behind the scenes with build.cmd
+
+- Setup tools (currently done in init-tools but will later be a boot-strap script in run.cmd/sh)
+- Restore external dependencies
+ - CoreCLR - Copy to bin\runtime\[BuildTarget]
+ - Netstandard Library - Copy to bin\ref\netstandard
+ - UAP - Copy to bin\runtime\uap??
+- Build targeting pack
+ - Build src\ref.builds file which will build all the references assemblies to bin\ref\[BuildTargetFramework]. Where BuildTargetFramework is a versionless TFM representing the
+ vertical build (e.g. netcoreapp/netfx/uap)
+- Build product
+ - Build src\src.builds file which will build all the source projects
+ - Source projects will have <Reference> to their dependencies or in some
+   case P2P references to other source projects when necessary (i.e. partial facades)
+ - Source projects will copy output to bin\runtime\[BuildConfiguration]
+- Tests will be built as part of build-tests.cmd and will require that build.cmd has
+  been ran at least once.
+- Packages will be built as part of build-packages.cmd and will require that build.cmd has
+  been run at least once.
+
+##Behind the scenes with build-test.cmd
+TODO:
+
+##Behind the scenes with build-packages.cmd
+TODO:
+
+#Build Pivots
+Below is a list of all the various options we pivot the project builds on.
 
-In the ref directory for the library there should be at most **one** `.csproj` that contains the latest API for the reference assembly for the library.  If a library cannot support all supported targets using the latest API it should use `.depproj` projects for each significant historical version in order to download and redistribute the historical reference assembly in the latest package.  If a library is a pure portable library with a single implementation it need not use a reference assembly at all. (see [Reference assembly projects](#reference-assembly-projects)).
+- **Flavor:** Debug, Release
+- **Architecture:** x86, x64, ARM, ARM64, AnyCPU
+- **OS:** Windows_NT, Linux, OSX, FreeBSD, AnyOS
+- **Target Frameworks:** NetFx (aka Desktop), netstandard (aka dotnet/Portable), NETCoreApp (aka .NET Core), UAP (aka UWP/Store/netcore50), [??Should we have any target framework??]
```

Comment:
Kind of but not exactly. I'm thinking of a scenario where your project configuration can say any for the TFM and we will essentially pick that configuration for what ever BuildConfiguration you are trying to build for. That will allow it to be specific for that vertical but not have to have duplicated configurations in the project.